lib/fakes3/sorted_object_list.rb

```diff
@@ -35,8 +35,9 @@ def find(object_name)
     def add(s3_object)
       return if !s3_object
 
-      @object_map[s3_object.name] = s3_object
-      @sorted_set << s3_object
+      if @sorted_set.add(s3_object)
+        @object_map[s3_object.name] = s3_object
+      end
```

Comment:
Wouldn't this logic have the effect of ignoring any subsequent setting of an `s3_object`? Wouldn't we instead want the last one added to "win"? If so, why not do something like the following?

```
@object_map[s3_object.name] = s3_object
@sorted_set.remove(s3_object)
@sorted_set << s3_object
```

Or does it do as you would expect the way you had it?